### PR TITLE
Fixes #3342

### LINKED ISSUE #3342: unspecified branch bonds in SMARTS don't have aromaticity set
Here's a demo:
image

This isn't a giant deal since the bond orders and aromaticity flags in reactions created from SMARTS are there as a convenience to make it easier to translate to SMILES, but it should still be fixed.

### PR DESCRIPTION
**Pull request by greglandrum**
This is a one-liner to set the appropriate property on the bond in `AddFragToMol()`

### CODE REVIEW

File: Code/GraphMol/SmilesParse/SmilesParseOps.cpp [hunk 1]
```diff
@@ -151,6 +151,7 @@ void AddFragToMol(RWMol *mol, RWMol *frag, Bond::BondType bondOrder,
         newB->setOwningMol(mol);
         newB->setBeginAtomIdx(atomIdx1);
         newB->setEndAtomIdx(atomIdx2);
+        newB->setProp(RDKit::common_properties::_unspecifiedOrder, 1);
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/SmilesParseOps.cpp:154]: @greglandrum interesting, is this related to #3307 by any chance?

Comment (greglandrum) [reply Code/GraphMol/SmilesParse/SmilesParseOps.cpp:154]: Unfortunately not. That seems to be connected to the aromaticity perception, not SMILES/SMARTS parsing
